src/gpu/utils.cu

```diff
@@ -29,11 +29,22 @@ extern "C" {
     }  
     return(0);
   }
-
   #ifdef __cplusplus
 }
 #endif
 
+#ifdef __cplusplus
+extern "C" {
+  #endif
+
+  int cudaresetdevice_bare(void) {
+    cudaDeviceReset();
```

Comment:
Should we do a `CUDACHECK` here? Also probably we can return `void` here instead?